src/renderer/app.tsx

```diff
@@ -672,8 +675,47 @@ export class App extends React.Component<AppProps> {
         log.error('Launcher', `Failed to register shortcut for ${shortcuts}, shortcut context missing?`);
       }
     });
-  }
 
+    window.Shared.back.register(BackOut.FPFSS_ACTION, async (event, extId: string) => {
+      return new Promise((resolve, reject) => {
+        const previousConsent = getFpfssConsentExt(extId);
+        if (previousConsent) {
+          this.performFpfssAction(async (user) => {
+            resolve(user);
+          });
+        } else if (previousConsent === false) {
+          reject(new Error('Extension is not allowed to access FPFSS token'));
+        } else {
+          remote.dialog.showMessageBox({
+            type: 'question',
+            title: 'FPFSS Extension Access',
+            message: `Extension with ID ${extId} is requesting access to your FPFSS token. Allow?`,
+            buttons: [this.context.misc.yes, this.context.misc.no],
+            defaultId: 1,
+            cancelId: 1,
+          })
+          .then(({ response }) => {
+            if (response === 0) {
+              this.performFpfssAction(async (user) => {
+                if (user) {
+                  saveFpfssConsentExt(extId, true);
+                  resolve(user);
+                } else {
+                  saveFpfssConsentExt(extId, false);

```

Comment:
We shouldn't save that the user didn't allow launcher to access FPFSS.
@colin969 agree?

Reply:
Probably for the best only to save approvals. If an extension repeatedly asks for permission and the user won't allow it, then it should be up to the extension to behave and the user can remove them.